royal/ext/sqla.py

```diff
@@ -0,0 +1,168 @@
+import logging
+
+from pyramid.location import lineage
+import royal
+from sqlalchemy.orm.collections import MappedCollection
+
+log = logging.getLogger(__name__)
+
+
+def includeme(config):
+    config.scan(__name__)
+
+
+class Collection(royal.Collection):
+    sa_model = None
+    sa_exceptions = None
+    entity_cls = None
+
+    def __init__(self, name, parent, entities=None):
+        super(Collection, self).__init__(name, parent)
+        self.entities = entities
+
+    def __repr__(self):
+        return '<%s collection at %s named %r>' % (self.__class__.__name__,
+                                                   id(self),
+                                                   self.name)
+
+    def load_entities(self):
+        self.entities = self.entity_cls.all()
+        # TODO pagination
+
+    def index(self, params):
+        if self.entities is None:
+            self.load_entities()
+        return self
+
+    def create(self, params):
+        entity = self.entity_cls(**params)
+        entity.save()
+        try:
+            self.sa_model.flush()
+        except self.sa_exceptions.DuplicatedEntity:
+            raise
```

Comment:
I know it's ugly, but we needed to raise an exception that's specific to the `sa` module injected..(we want to differentiate between types of conflicts)... On a second thought, this particular line is not really needed since we're re-raising all exceptions anyway, there was no point catching this particular exception..